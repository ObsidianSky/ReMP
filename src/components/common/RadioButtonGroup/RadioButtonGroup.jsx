import React, { Component } from 'react';
import PropTypes from 'prop-types';
if (process.env.BROWSER) {
	require('./RadioButtonGroup.scss');
}

import Button from '../Button/Button';

const classMap = {
    search: 'btn--narrow btn--gray',
    sort: 'btn--link'
};

class RadioButtonGroup extends Component {
    getClass(option) {
        const classes = classMap[this.props.type];

        return this.isActive(option)
            ? `${classes} active`
            : classes;
    }

    isActive(option) {
        return option.value === this.props.activeOption;
    }

    selectOption(option) {
        this.props.onSelect(option);
    }

    getOptions() {
        return this.props.options
            .map((option, index) =>
                <Button className={this.getClass(option)}
                        onClick={() => this.selectOption(option.value)}
                        key={index}
                >
                    {option.label}
                </Button>
            )
    }

    render() {
        return (
            <div className="radio-button-group">
                <div className={`radio-button-group__heading radio-button-group__heading--${this.props.type}`}>
                    {this.props.title}
                </div>
                <div className="radio-button-group__options">
                    {this.getOptions()}
                </div>
            </div>
        )
    }
}

RadioButtonGroup.defaultProps = {
    title: '',
    type: 'search'
};

RadioButtonGroup.propTypes = {
    title: PropTypes.string,
    type: PropTypes.oneOf(['search', 'sort']),
    onSelect: PropTypes.func.isRequired,
    options: PropTypes.arrayOf(PropTypes.shape({
        label: PropTypes.string,
        value: PropTypes.string
    })).isRequired,
    activeOption: PropTypes.string
};

export default RadioButtonGroup;