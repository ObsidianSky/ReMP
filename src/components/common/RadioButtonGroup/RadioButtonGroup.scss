@import '../../../assets/styles/variables';

.radio-button-group {
  display: flex;
  align-items: center;

  &__heading {
    color: $text-light;
    text-transform: uppercase;
    font-size: 15px;
    margin-right: 15px;

    &--sort {
      text-transform: none;
      color: $text-black;
      font-weight: 600;
    }
  }

  &__options {
    > * + * {
      margin-left: 12px;
    }
  }
}