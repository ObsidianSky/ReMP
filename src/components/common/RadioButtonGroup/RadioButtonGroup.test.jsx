import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import RadioButtonGroup from './RadioButtonGroup';

describe('RadioButtonGroup', () => {
    let props;

    beforeEach(() => {
        props = {
            title: 'title',
            type: 'search',
            onSelect: () => {},
            options: [
                {
                    label: 'label1',
                    value: 'value1'
                }, {
                    label: 'label2',
                    value: 'value2'
                }
            ],
            activeOption: 'value2'
        }
    });

    describe('when it is search radio button group', () => {
        test('should renders correctly', () => {
            const wrapper = shallow(<RadioButtonGroup { ...props } />);

            expect(toJson(wrapper)).toMatchSnapshot();
        });
    });

    describe('when it is sort radio button group', () => {
        beforeEach(() => {
            props.type = 'sort';
        });

        test('should renders correctly', () => {
            const wrapper = shallow(<RadioButtonGroup { ...props } />);

            expect(toJson(wrapper)).toMatchSnapshot();
        });
    });

    describe('when there is no active option', () => {
        beforeEach(() => {
            props.activeOption = '';
        });

        test('should renders correctly', () => {
            const wrapper = shallow(<RadioButtonGroup { ...props } />);

            expect(toJson(wrapper)).toMatchSnapshot();
        });
    });
});