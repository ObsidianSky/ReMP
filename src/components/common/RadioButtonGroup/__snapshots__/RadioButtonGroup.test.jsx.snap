// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButtonGroup when it is search radio button group should renders correctly 1`] = `
<div
  className="radio-button-group"
>
  <div
    className="radio-button-group__heading radio-button-group__heading--search"
  >
    title
  </div>
  <div
    className="radio-button-group__options"
  >
    <Button
      className="btn--narrow btn--gray"
      key="0"
      onClick={[Function]}
      text=""
    >
      label1
    </Button>
    <Button
      className="btn--narrow btn--gray active"
      key="1"
      onClick={[Function]}
      text=""
    >
      label2
    </Button>
  </div>
</div>
`;

exports[`RadioButtonGroup when it is sort radio button group should renders correctly 1`] = `
<div
  className="radio-button-group"
>
  <div
    className="radio-button-group__heading radio-button-group__heading--sort"
  >
    title
  </div>
  <div
    className="radio-button-group__options"
  >
    <Button
      className="btn--link"
      key="0"
      onClick={[Function]}
      text=""
    >
      label1
    </Button>
    <Button
      className="btn--link active"
      key="1"
      onClick={[Function]}
      text=""
    >
      label2
    </Button>
  </div>
</div>
`;

exports[`RadioButtonGroup when there is no active option should renders correctly 1`] = `
<div
  className="radio-button-group"
>
  <div
    className="radio-button-group__heading radio-button-group__heading--search"
  >
    title
  </div>
  <div
    className="radio-button-group__options"
  >
    <Button
      className="btn--narrow btn--gray"
      key="0"
      onClick={[Function]}
      text=""
    >
      label1
    </Button>
    <Button
      className="btn--narrow btn--gray"
      key="1"
      onClick={[Function]}
      text=""
    >
      label2
    </Button>
  </div>
</div>
`;
